src/repositories/user.repository.ts

```diff
@@ -41,4 +42,23 @@ export const addUserData = async (user: User) => {
   }
 };
 
+export const findUser = async (user: User) => {
+  const { username, password } = user;
+
+  if (user && username && password) {
+    const getUserPassword = await UserModel.findOne({
+      username: username,
+    }).select("password");
+    if (!comparePassword(getUserPassword.password, password)) {
```

Comment:
this logic should not be in repository, because it's "logic". All logic should be on service level